widget/progressbar.go

```diff
@@ -114,15 +122,55 @@ func (p *ProgressBar) SetValue(v float64) {
 	Renderer(p).Refresh()
 }
 
+// Stops the infinite progress goroutine and sets value to the Max
+func (p *ProgressBar) StopInfiniteProgress() {
+	if p.stopInfiniteLoopChan != nil {
+		p.stopInfiniteLoopChan <- true
+	}
+	p.SetValue(p.Max)
+}
+
+// internal loop called with `go infiniteProgressLoop()`
+// updates the infinite-style progress bar
+// can be exited by calling ProgressBar.StopInfiniteProgress()
+func (p *ProgressBar) infiniteProgressLoop() {
+	if p.stopInfiniteLoopChan == nil {
+		log.Println("No way to stop infinite progress loop, so not allowing to start.")
+		return
+	}
+
+	tickChan := time.NewTicker(infiniteRefreshRate).C
```

Comment:
ticker := time.NewTicker(infiniteRefreshRate)
defer ticker.Stop()

then use `case <-ticker.C` below

Otherwise - Im pretty sure ... each call to infiniteProgressLoop() will leave a goroutine running that continues to emit ticks.   eeeks !